store/store.go

```diff
@@ -374,6 +385,32 @@ func (s *Store[H]) deleteRange(ctx context.Context, batch datastore.Batch, from,
 	return nil
 }
 
+func (s *Store[H]) updateTail(ctx context.Context, from, to uint64) error {
+	tailPtr := s.tailHeader.Load()
+	if tailPtr == nil {
+		return nil
+	}
+
+	tailH := (*tailPtr).Height()
+	// tail is far below from - we will update it in the future
+	// tail is far above to - we are removing irrelevant headers
+	if tailH < from || to < tailH {
+		return nil
+	}
+
+	for h := to; ; h++ {
+		newTail, err := s.getByHeight(ctx, h)
+		if err == nil {
+			s.tailHeader.Store(&newTail)
```

Comment:
we should also change the tail pointer on disk
to be added together with delete batch

Reply:
We need to merge Tail first.

Reply:
:shipit:

Reply:
Done